skipchain/struct.go

```diff
@@ -1027,3 +1112,57 @@ func (db *SkipBlockDB) getAllSkipchains() (map[string]*SkipBlock, error) {
 	}
 	return gen, nil
 }
+
+// skipBlockBuffer will cache a proposed block when the conode has
+// verified it and it will later store it in the DB after the protocol
+// has succeeded.
+// Blocks are stored per skipchain to prevent the cache to grow and it
+// is cleared after a new block is added (again per skipchain)
+type skipBlockBuffer struct {
+	buffer map[[32]byte]*SkipBlock
+	sync.Mutex
+}
+
+func newSkipBlockBuffer() *skipBlockBuffer {
+	return &skipBlockBuffer{
+		buffer: make(map[[32]byte]*SkipBlock),
```

Comment:
We're using `string` as key in other places - @jeffallen can surely tell us which one is better: using `string` as key or array of bytes?